src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -979,6 +979,16 @@ public static bool SequenceEqual<TSource>(this IEnumerable<TSource> first, IEnum
             if (comparer == null) comparer = EqualityComparer<TSource>.Default;
             if (first == null) throw Error.ArgumentNull("first");
             if (second == null) throw Error.ArgumentNull("second");
+
+            if (first == second) return true;
```

Comment:
Well, one doesn't generally do it on purpose as by-the-by as part of a series of comparisons, but I think that case too is probably unlikely here. And if it is done, it's probably as part of some more general equality check that should probably include the identity-check itself. I'm going to take this bit out and squash it to a commit with just the collection test.